openupgrade_scripts/scripts/website_slides/16.0.2.6/pre-migration.py

```diff
@@ -0,0 +1,107 @@
+from openupgradelib import openupgrade
+
+renamed_fields = [
+    ("slide.slide", "slide_slide", "datas", "binary_content"),
+    ("slide.channel", "slide_channel", "share_template_id", "share_slide_template_id"),
+]
+
+xml_ids_to_rename = [
+    (
+        "website_slides.rule_slide_slide_resource_manager",
+        "website_slides.rule_slide_slide_resource_downloadable_manager",
+    ),
+]
+
+# slide_slide
+
+
+def create_and_fill_data_from_slide_type_to_slide_category(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE slide_slide
+        ADD COLUMN IF NOT EXISTS slide_category VARCHAR
+        """,
+    )
+
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE slide_slide
+        SET slide_category = CASE
+            WHEN slide_type = 'webpage' THEN 'article'
+            WHEN slide_type = 'presentation' THEN 'document'
+            ELSE slide_type
+        END
+        """,
+    )
+
+
+def create_and_fill_data_for_source_type(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE slide_slide
+        ADD COLUMN IF NOT EXISTS source_type VARCHAR
+        """,
+    )
+
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE slide_slide
+        SET source_type = CASE
+            WHEN url IS NOT NULL AND slide_category = 'document' THEN 'external'
```

Comment:
@nguyenvietlam0640 chỉ cần như này là đủ
tại sao cần if thêm slide_category = 'document'?
```suggestion
            WHEN url IS NOT NULL THEN 'external'
```

Reply:
tại vì `slide_category = 'video'` cũng có url anh ơi. Thêm nữa là  `source_type `là  `external` hay `local_file` chỉ dùng với thằng `slide_category = 'document'`